src/pyroboplan/ik/differential_ik.py

```diff
@@ -157,6 +163,26 @@ def solve(
         np.random.seed(self.options.rng_seed)
         target_frame_id = self.model.getFrameId(target_frame)
 
+        # Get the active joint indices.
+        active_joint_indices = [
+            idx
+            for idx in range(self.model.nq)
+            if idx not in self.options.ignore_joint_indices
+        ]
+        num_active_joints = len(active_joint_indices)
+
+        # Create the joint weights.
+        if self.options.joint_weights is None:
+            # Use identity weights if they are not specified.
+            W = np.eye(num_active_joints)
+        elif len(self.options.joint_weights) != num_active_joints:
+            raise ValueError(
+                f"Joint weights, if specified, must have {num_active_joints} elements."
+            )
+        else:
+            # Invert the weights so that higher weight means less joint motion.
+            W = np.linalg.inv(np.diag(self.options.joint_weights))
```

Comment:
Do you want to add any kind of validation here? Like you can't have 0 as a weight or this becomes singular. Also, what does a negative weight mean?

Reply:
Good idea.